### PR TITLE
feat(ctb): add message versioning to MessagePasser

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Adds message versioning to the L2ToL1MessagePasser, using the same scheme as in the CrossDomainMessenger. Also renames the "nonce" variable to "messageNonce" in line with the naming within the CrossDomainMessenger contracts.

### CODE REVIEW

File: packages/contracts-bedrock/contracts/L2/L2ToL1MessagePasser.sol
Comment (tynes) [inline packages/contracts-bedrock/contracts/L2/L2ToL1MessagePasser.sol:103]: I think you should also add logic to `Hashing.hashWithdrawal` that observes the version

File: packages/contracts-bedrock/contracts/L2/L2ToL1MessagePasser.sol
Comment (tynes) [inline packages/contracts-bedrock/contracts/L2/L2ToL1MessagePasser.sol:117]: The migrated withdrawals will be a special case, where they do not have the version applied to the nonce, but they are hashing with version 1 style hashing

File: packages/contracts-bedrock/contracts/L2/L2ToL1MessagePasser.sol
Comment (smartcontracts) [reply packages/contracts-bedrock/contracts/L2/L2ToL1MessagePasser.sol:103]: Feels unnecessary for now since we don't actually have a second version. It would be easy to add in the future if/when we have a second version.

File: packages/contracts-bedrock/contracts/L2/L2ToL1MessagePasser.sol
Comment (smartcontracts) [reply packages/contracts-bedrock/contracts/L2/L2ToL1MessagePasser.sol:117]: That's fine, we just say that version 0 and 1 have the same hashing scheme

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
I don't think that there are any implications of this PR, but would like a review from @maurelian before merge just to get another set of eyes on this.

The contract didn't exist previously, there should be no nonce collisions with migrated withdrawals and the nonces should be guaranteed to be monotonic

**Review by maurelian — APPROVED** [review]

**Comment by tynes** [comment]
@mergifyio requeue

**Pull request merged**
